src/Symfony/Component/BrowserKit/Tests/ClientTest.php

```diff
@@ -344,6 +354,17 @@ public function testSubmit()
         $this->assertEquals('http://www.example.com/foo', $client->getRequest()->getUri(), '->submit() submit forms');
     }
 
+    public function testSubmitForm()
+    {
+        $client = new TestClient();
+        $client->setNextResponse(new Response('<html><form action="/foo"><input type="submit" value="Register" /></form></html>'));
+        $client->request('GET', 'http://www.example.com/foo/foobar');
+
+        $client->submitForm('Register', array(), 'POST');
```

Comment:
I think it would be nice to add a test case where actual values are submitted along, this case is not covered right now.